### ISSUE TITLE
UDP Listen Flow Example doesn't work

### ISSUE THREAD

**Issue by DavidSchinazi** [tfpauly/draft-age-masque-connect-ip#9]
I don't understand how the UDP Listen Flow Example works. Unless I'm missing something, for this example to work you would need a way to convey the client's UDP port number. Since we decided that this was out of scope, perhaps we should remove this example?

**Comment by tfpauly** [comment]
This example was more about saying we can do UDP listening on an address, but all ports. So, it's binding just the protocol and address. It is something that is begging for a port extension to scope down further, but it still shows how you can do protocol listening, without being a full IP tunnel. Essentially, saying "I only implement user space UDP and can handle that, I don't want anything else".

**Comment by DavidSchinazi** [comment]
This is the first time I've ever heard of "UDP listening on all ports". I'm not saying it isn't useful, but perhaps our examples should be focused on use-cases that people have a clear use for?

**Comment by gloinul** [comment]
So our draft had mechanism for the client to specify traffic selectors for what it actually considers to be individual IP flows. Think about how the rules to create IPv6 flow label works. MASQUE would need something that allows the client to indicate the necessary traffic selector that results in a IP flow, without having the local API interaction that for examples enables an stack to convert sockets into flows. I think this is the question we have here. And I don't think IP proto/next header is sufficient selector here.

**Comment by achernya** [comment]
RFC 6437 seems to say "Packet classifiers can use the triplet of Flow Label, Source Address, and Destination Address" which suggests to me that source address + destination address are the scopes within which flow label is evaluated. As a result, I don't see what changes we'd need here to support flow labels? It seems fine to pass them through unmodified here. If some extension wanted to add further traffic selectors that limit things by flow label, that would be possible without adding anything to the base draft.

**Comment by gloinul** [comment]
@achernya yes, for an on path device that does packet classification the flow label is scoped by the source and destination address. And that would be true in the case one only encapsulates whole IPv6 packets. However, as soon as the MASQUE endpoint becomes more flow aware one have to discuss a bit more on how this flow awareness is created. In endpoints the setting of IPv6 flow label is clearly created based on transport protocol flows. Also IPv6 flow labels could be a selector that would only work for IPv6. There is no IPv4 equivalent, and also what to do with zero flow label becomes interesting.
